packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -67,196 +63,12 @@ public class GoLLMController {
-					// add provenance
-					provenanceService.createProvenance(new Provenance()
-							.setLeft(newConfig.getId())
-							.setLeftType(ProvenanceType.MODEL_CONFIGURATION)
-							.setRight(props.documentId)
-							.setRightType(ProvenanceType.DOCUMENT)
-							.setRelationType(ProvenanceRelationType.EXTRACTED_FROM));
-				}
-
-			} catch (final Exception e) {
-				log.error("Failed to configure model", e);
-				throw new RuntimeException(e);
-			}
-			log.info("Model configured successfully");
-		});
-		return handler;
-	}
-
-	private TaskResponseHandler configureFromDatasetHandler() {
-		final TaskResponseHandler handler = new TaskResponseHandler();
-		handler.onSuccess((TaskResponse resp) -> {
-			try {
-				final ConfigFromDatasetProperties props = resp
-						.getAdditionalProperties(ConfigFromDatasetProperties.class);
-
-				final Model model = modelService.getAsset(props.getModelId())
-						.orElseThrow();
-				final ConfigureModelResponse configurations = objectMapper.readValue(resp.getOutput(),
-						ConfigureModelResponse.class);
-
-				// For each configuration, create a new model configuration with parameters set
-				for (JsonNode condition : configurations.response.get("conditions")) {
-					// Map the parameters values to the model
-					final Model modelCopy = new Model(model);
-					final List<ModelParameter> modelParameters = modelCopy.getSemantics().getOde().getParameters();
-					modelParameters.forEach((parameter) -> {
-						JsonNode conditionParameters = condition.get("parameters");
-						conditionParameters.forEach((conditionParameter) -> {
-							if (parameter.getId().equals(conditionParameter.get("id").asText())) {
-								parameter.setValue(conditionParameter.get("value").doubleValue());
-							}
-						});
-					});
-
-					// Create the new configuration
-					final ModelConfiguration configuration = new ModelConfiguration();
-					configuration.setModelId(model.getId());
-					configuration.setName(condition.get("name").asText());
-					configuration.setDescription(condition.get("description").asText());
-					configuration.setConfiguration(modelCopy);
-
-					for (UUID datasetId : props.datasetIds) {
-						final ModelConfiguration newConfig = modelConfigurationService.createAsset(configuration);
-						// add provenance
-						provenanceService.createProvenance(new Provenance()
-								.setLeft(newConfig.getId())
-								.setLeftType(ProvenanceType.MODEL_CONFIGURATION)
-								.setRight(datasetId)
-								.setRightType(ProvenanceType.DATASET)
-								.setRelationType(ProvenanceRelationType.EXTRACTED_FROM));
-					}
-				}
-
-			} catch (Exception e) {
-				log.error("Failed to configure model", e);
-				throw new RuntimeException(e);
-			}
-			log.info("Model configured successfully");
-		});
-		return handler;
+		taskService.addResponseHandler(new ModelCardResponseHandler(objectMapper, documentAssetService));
```

Comment:
We should autowire these as private members of the GoLLMController so we don't need to manually pass the deps. Just add @Component to the class definitions and the @RequiredArgsContructor stuff

Reply:
I was thinking there might be some "nice" refactoring beyond what I have done, but as I cannot test anything currently I was thinking this might be good enough for now?